twine/repository.py

```diff
@@ -132,7 +142,7 @@ def _upload(self, package):
                 allow_redirects=False,
                 headers={'Content-Type': monitor.content_type},
             )
-            bar.done()
+            bar.close()
```

Comment:
If not it might be worth it to do something like this:
```python
bar = ProgressBar(total=encoder.len, unit='bytes', unit_scale=True, leave=False)
try:
    # Setup monitor and response here.
finally:
    bar.close()
```